tests/PySys/mqtt_port/mqtt_port_change_connection_works/run.py

```diff
@@ -0,0 +1,132 @@
+import sys
+
+sys.path.append("environments")
+from environment_roundtrip_c8y import Environment_roundtrip_c8y
+
+from pysys.basetest import BaseTest
+
+import time
+
+"""
+Validate changing the mqtt port using the tedge command
+
+Given a configured system, that is configured with certificate created and registered in a cloud
+When the thin edge device is disconnected from cloud, sudo tedge disconnect c8y/az
+When mqtt.port is set using tedge with sudo
+When the thin edge device is connected to cloud, sudo tedge connect c8y/az
+Now validate the services that use the mqtt port
+   Validate tedge config set mqtt.port
+   Validate tedge mqtt pub/sub
+   Validate tedge connect c8y/az --test
+   Validate tedge_mapper status
+
+"""
+
+class MqttPortChangeConnectionWorks(Environment_roundtrip_c8y):
+    def setup(self):
+        self.tedge = "/usr/bin/tedge"
+        self.sudo = "/usr/bin/sudo"
+
+        # disconnect from c8y cloud
+        disconnect_c8y = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "disconnect", "c8y"],
+            stdouterr="disconnect_c8y",
+        )
+        self.addCleanupFunction(self.mqtt_cleanup)
+
+    def execute(self):
+        # set a new mqtt port for local communication
```

Comment:
The Environment_roundtrip_c8y already has a execute and validation function, that would run a smoketest and validate. By overriding that methods here, the methods from the baseclass are not called anymore. Using EnvironmentC8y would probably be better here.

Reply:
If you still want to call those methods you can call 
```python
super().execute()
```